.ci/write-dockerfile.sh

```diff
@@ -349,14 +349,9 @@ ENV SAGE_CHECK_PACKAGES="!cython,!python3,!cysignals,!linbox,!ppl,!cmake,!rpy2,!
 $ADD .gitignore /new/.gitignore
 $ADD src /new/src
 RUN cd /new && rm -rf .git && \\
-    if /.ci/retrofit-worktree.sh worktree-pre /sage; then \\
-        cd /sage && touch configure build/make/Makefile; \\
-    else \\
-        echo "retrofit-worktree.sh failed, falling back to replacing /sage/src"; \\
-        rm -rf /sage/src; \\
-        mv src /sage/src; \\
-        cd /sage && ./bootstrap && ./config.status; \\
-    fi; \\
+    rm -rf /sage/src; \\
+    mv src /sage/src; \\
+    cd /sage && ./bootstrap && ./config.status; \\
```

Comment:
By the way what is this part supposed to do? As I read retrofit-worktree does

```
    echo >&2 "usage: $0 WORKTREE_NAME WORKTREE_DIRECTORY"
    echo >&2 "Ensures that the current working directory is a git repository,"
    echo >&2 "then makes WORKTREE_DIRECTORY a git worktree named WORKTREE_NAME."
```

but I don't understand why. Your commit would make sense if you assume the retrofit-worktree always fail (but why would it?)

Reply:
retrofit-worktree is to perform "incremental build" from the previous stage.

The change here makes ""incremental build" broken, and thus also the whole point of the multi-stage docker builds.

By the way, I made changes to simplify the multi-stage docker builds in the other PR.